### ISSUE TITLE
Inversion goes to collapse

### ISSUE THREAD

**Issue by ChengBinJin** [genforce/idinvert#21]
@zhujiapeng @ShenYujun @zhoubolei 
I like your works including InterFaceGAN.

I test some FFHQ images (from 00000.png to 00029.png, 30 images) using the In-Domain GAIN Inversion method. I found for some examples, the output of the encoder seems good but during some iterations that the results do not seem correct. The following two images using 100 iterations and 1,000 iterations. But the result even becomes worse. For example, the problem of 00000.png was corrected during long iterations, but 00002.png no big difference. In addition, the 00015.png becomes worse.
Do you have some ideas about this?

![100_iters](https://user-images.githubusercontent.com/37034031/93855928-e4d60700-fcea-11ea-9251-208041aac272.png)

![1000_iters](https://user-images.githubusercontent.com/37034031/93855939-e9022480-fcea-11ea-8e4e-f05ef923b117.png)

**Comment by zhujiapeng** [comment]
In this case, you could decrease the `--learning_rate` (i.e., ten times smaller) in `invert.py`. By the way, we have updated the encoder network and its weight in the repo, you could try it without adjusting the learning rate.

**Comment by ChengBinJin** [comment]
@zhujiapeng 
After updating the new encoder network and its weight, I got the following error. Did you encounter this problem?
ModuleNotFoundError: No module named 'tensorflow.contrib.nccl

![image](https://user-images.githubusercontent.com/37034031/93875388-794e6280-fd07-11ea-8a82-ac9f9c949ba4.png)

**Comment by zhujiapeng** [comment]
We added Synchronized BatchNorm in the encoder, it will use `nccl `  in TensorFlow, the tf version I trained encoder is 1.12, so you need to change the tf version to 1.12.

**Comment by ChengBinJin** [comment]
@zhujiapeng 
Thank you for your quickly reply. I currently use TensorFlow 1.14.0, I will try it on 1.12. 👍

**Comment by ChengBinJin** [comment]
@zhujiapeng 
I success to run the new model using TF 1.12.2 as your suggestion.
Even though the image, 00015.png still has a similar problem, for others the new model improves a lot. Great work! I will close this issue.

![iter_100](https://user-images.githubusercontent.com/37034031/93958040-2963b000-fd88-11ea-8a00-d59d196b9452.png)

![iter_1000](https://user-images.githubusercontent.com/37034031/93958045-2cf73700-fd88-11ea-8d22-d2fcf9956587.png)

**Comment by zhujiapeng** [comment]
@ChengBinJin 
I picked the wrong model when uploading, I have updated it, sorry about that.
The new model's inversion results show as follows, but this problem may happen on some other images, this caused by the small batch size when training the encoder due to our GPU memory.  Anyway, thank you for testing our model!
![inv](https://user-images.githubusercontent.com/10594850/93966718-f3c6c300-fd97-11ea-98c1-1cc57a25993e.png)
